ipykernel/debugger.py

```diff
@@ -442,9 +442,7 @@ async def source(self, message):
                     'content': f.read()
                 }
         else:
-            reply['success'] = False
-            reply['message'] = 'source unavailable'
-            reply['body'] = {}
+            reply = await self._forward_message(message)
```

Comment:
See my [comment](https://github.com/ipython/ipykernel/issues/870#issuecomment-1049008052); I think the right implementation should be to handle the request in the kernel if and only if the attribute `sourceReference` is 0. Notice that the `source` attribute is optional, accoriding to the [DAP](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Source), therefore this implementation is not totally compliant.

Reply:
I've had another go at this, trying to follow the spec more closely this time.